lib/cdo/mailjet.rb

```diff
@@ -57,29 +61,33 @@ def self.create_contact(email, name, sign_up_date)
     end
 
     contact = Mailjet::Contactdata.find(email)
+    return contact if contact&.id.present?
 
-    if contact&.id.nil?
-      Mailjet::Contact.create(
-        is_excluded_from_campaigns: true,
-        email: email,
-        name: name
-      )
-      contact = Mailjet::Contactdata.find(email)
-    end
+    Mailjet::Contact.create(
+      is_excluded_from_campaigns: true,
+      email: email,
+      name: name
+    )
+    Mailjet::Contactdata.find(email)
+  end
+
+  def self.update_contact_field(contactdata, field_name, field_value)
+    return unless enabled?
+    return if contactdata.nil?
 
-    sign_up_date_rfc3339 = sign_up_date.rfc3339
-    contact.update_attributes(
+    contactdata.update_attributes(
       data: [
         {
-          name: 'sign_up_date',
-          value: sign_up_date_rfc3339
+          name: field_name,
+          value: field_value
         }
       ]
     )
   end
 
-  def self.send_template_email(to_email, to_name, email_config)
+  def self.send_template_email(contactdata, email_config)
```

Comment:
Is there any world in which this returns an error?

Reply:
I'm trying to remember -- I actually had more issues with this failing silently instead of throwing errors. Do you think this should be wrapped in exception handling in some way? (Might also be feedback for me to incorporate into https://github.com/code-dot-org/code-dot-org/pull/58099)